source/run_time_parameters.cc

```diff
@@ -509,6 +509,222 @@ Stream& operator<<(Stream &stream, const DimensionlessNumbers &prm)
+    convective_term_time_discretization = ConvectiveTermTimeDiscretization::semi_implicit;
+  else if (str_convective_term_time_discretization == std::string("explicit"))
+    convective_term_time_discretization = ConvectiveTermTimeDiscretization::fully_explicit;
+  else
+    AssertThrow(false,
+                ExcMessage("Unexpected identifier for the time discretization "
+                            "of the convective term."));
+
+  preconditioner_update_frequency = prm.get_integer("Preconditioner update frequency");
+  AssertThrow(preconditioner_update_frequency > 0,
+          ExcLowerRange(preconditioner_update_frequency, 0));
+
+  verbose = prm.get_bool("Verbose");
+
+  prm.enter_subsection("Linear solver parameters - Diffusion step");
+  {
+    diffusion_step_solver_parameters.parse_parameters(prm);
+  }
+  prm.leave_subsection();
+
+  prm.enter_subsection("Linear solver parameters - Projection step");
+  {
+    projection_step_solver_parameters.parse_parameters(prm);
+  }
+  prm.leave_subsection();
+
+  prm.enter_subsection("Linear solver parameters - Correction step");
+  {
+    correction_step_solver_parameters.parse_parameters(prm);
+  }
+  prm.leave_subsection();
+
+  prm.enter_subsection("Linear solver parameters - Zeroth step");
+  {
+    zeroth_step_solver_parameters.parse_parameters(prm);
+  }
+  prm.leave_subsection();
+}
+
+
+
+template<typename Stream>
+Stream& operator<<(Stream &stream, const ProjectionSolverParametersBase &prm)
+{
+  switch (prm.convective_term_weak_form) {
+    case ConvectiveTermWeakForm::standard:
+      internal::add_line(stream, "Convective term weak form", "standard");
+      break;
+    case ConvectiveTermWeakForm::rotational:
+      internal::add_line(stream, "Convective term weak form", "rotational");
+      break;
+    case ConvectiveTermWeakForm::divergence:
+      internal::add_line(stream, "Convective term weak form", "divergence");
+      break;
+    case ConvectiveTermWeakForm::skewsymmetric:
+      internal::add_line(stream, "Convective term weak form", "skew-symmetric");
+      break;
+    default:
+      AssertThrow(false, ExcMessage("Unexpected type identifier for the "
+                               "weak form of the convective term."));
+      break;
+  }
+
+  switch (prm.convective_term_time_discretization) {
+    case ConvectiveTermTimeDiscretization::semi_implicit:
+      internal::add_line(stream, "Convective temporal form", "semi-implicit");
+      break;
+    case ConvectiveTermTimeDiscretization::fully_explicit:
+      internal::add_line(stream, "Convective temporal form", "explicit");
+      break;
+    default:
+      AssertThrow(false,
+                  ExcMessage("Unexpected type identifier for the "
+                             "time discretization of the convective term."));
+      break;
+  }
+
+  internal::add_line(stream, "Preconditioner update frequency", prm.preconditioner_update_frequency);
+
+  stream << prm.diffusion_step_solver_parameters;
+
+  stream << "\r";
```

Comment:
I am not a fan of these carriage returns because they result in different output on different machines. This makes it hard to compare results and to program tests.

Reply:
Okay I meant `"\r"`.